hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/repository/EntityStakeRepository.java

```diff
@@ -69,25 +68,31 @@ select lower(timestamp_range) as timestamp from entity_history where id = 800
           limit 1
         ), entity_state as (
           select
-            decline_reward,
             id,
             staked_account_id,
             staked_node_id,
             stake_period_start
           from entity, end_period
-          where deleted is not true and type in ('ACCOUNT', 'CONTRACT') and timestamp_range @> end_period.consensus_timestamp
+          where id = 800 or
+            (deleted is not true and type in ('ACCOUNT', 'CONTRACT') and timestamp_range @> end_period.consensus_timestamp
+                and
+              ((decline_reward is false and (staked_node_id <> -1 or staked_account_id <> 0)) or (decline_reward is true and staked_account_id <> 0))
```

Comment:
rewrite the condition so it's more stragithforward
```suggestion
              (staked_account_id <> 0 or (decline_reward is false and staked_node_id <> -1))
```

do the same for the `entity_history` part

Reply:
Thanks I have updated this.